kv/kvserver: TestStoreRangeSplitRaceUninitializedRHS failed

kv/kvserver.TestStoreRangeSplitRaceUninitializedRHS [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3084906&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3084906&tab=artifacts#/) on master @ [ba16357c4968ec68b2d0afda7862331f22e14e8c](https://github.com/cockroachdb/cockroach/commits/ba16357c4968ec68b2d0afda7862331f22e14e8c):


```
=== RUN   TestStoreRangeSplitRaceUninitializedRHS
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeSplitRaceUninitializedRHS222619826
    test_log_scope.go:74: use -show-logs to present logs inline
    testcluster.go:335: pq: setting updated but timed out waiting to read new value
    panic.go:636: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeSplitRaceUninitializedRHS222619826
W210615 14:20:48.009869 977027 server/node.go:783  [n1,summaries] 13  health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:ranges.underreplicated Value:37}]}
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (20.95s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestStoreRangeSplitRaceUninitializedRHS PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/kv tbg
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeSplitRaceUninitializedRHS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-8060

kv/kvserver.TestStoreRangeSplitRaceUninitializedRHS [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3137141&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3137141&tab=artifacts#/) on master @ [40b50c952aaa90e0ac565c859c1076a77c9a64be](https://github.com/cockroachdb/cockroach/commits/40b50c952aaa90e0ac565c859c1076a77c9a64be):


```
=== RUN   TestStoreRangeSplitRaceUninitializedRHS
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeSplitRaceUninitializedRHS849579772
    test_log_scope.go:74: use -show-logs to present logs inline
    testcluster.go:334: pq: setting updated but timed out waiting to read new value
    panic.go:632: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeSplitRaceUninitializedRHS849579772
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (18.98s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestStoreRangeSplitRaceUninitializedRHS PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/kv tbg
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeSplitRaceUninitializedRHS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/client_split_test.go
+++ b/pkg/kv/kvserver/client_split_test.go
@@ -50,7 +50,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/ts"
@@ -2025,7 +2024,6 @@ func TestStoreSplitGCHint(t *testing.T) {
 // and the uninitialized replica reacting to messages.
 func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 66480, "flaky test")
 	defer log.Scope(t).Close(t)
 
 	currentTrigger := make(chan *roachpb.SplitTrigger, 1)
@@ -2103,6 +2101,7 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 
 	for i := 0; i < 10; i++ {
 		errChan := make(chan *kvpb.Error)
+		failedSendLog := log.Every(time.Second)
 
 		// Closed when the split goroutine is done.
 		splitDone := make(chan struct{})
@@ -2147,7 +2146,15 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 						Term: term,
 					},
 				}, rpc.DefaultClass); !sent {
-					t.Error("transport failed to send vote request")
+					// SendAsync can return false, indicating the message didn't send.
+					// The most likely reason this test encounters a message failing to
+					// send is the outgoing message queue being full. The queue filling
+					// up is expected given it has fixed capacity and this loop is
+					// attempting to sending 1 MsgVote every microsecond. See comments
+					// below and above for the frequency rationale.
+					if failedSendLog.ShouldLog() {
+						log.Infof(ctx, "transport failed to send vote request")
+					}
 				}
 				select {
 				case <-splitDone:

